html/semantics/forms/the-select-element/common-HTMLOptionsCollection-namedItem.html

```diff
@@ -37,26 +39,16 @@
 }, "if no item has a name or id value matching the parameter, return null and stop");
 
 test(function () {
-    var testarr = [];
-    for (var i = 0; i < selly.namedItem('id3').length; i++) {
-        testarr.push(selly.namedItem('id3')[i].text);
-    }
-    assert_array_equals(testarr, ['3','duplicate ID']);
-}, "return an HTMLOptionsCollection in correct order for repeated 'id' value");
+    assert_equals(selly.namedItem('id3')["value"], "3");
+}, "if multiple items have a name or *id* value matching the parameter, return the first object and stop");
 
 test(function () {
-    var testarr = [];
-    for (var i = 0; i < selly.namedItem('name4').length; i++) {
-        testarr.push(selly.namedItem('name4')[i].text);
-    }
-    assert_array_equals(testarr, ['4', 'duplicate name']);
-}, "return an HTMLOptionsCollection in correct order for repeated 'name' value");
+    assert_equals(selly.namedItem('name4')["value"], "4");
+}, "if multiple items have a *name* or id value matching the parameter, return the first object and stop");
 
 test(function () {
-    var testarr = [];
-    for (var i = 0; i < selly.namedItem('mixed1').length; i++) {
-        testarr.push(selly.namedItem('mixed1')[i].text);
-    }
-    assert_array_equals(testarr, ['mixed ID', 'mixed name']);
-}, "return an HTMLOptionsCollection in correct order for repeated mixed value");
+    assert_equals(selly.namedItem('mixed1')["value"], "mixed ID");
```

Comment:
OK, thanks. I've added Ryosuke's test. WebKit passes, but Firefox fails -- it finds the element with ```name="mixed2"```.

Following on, I have a question about the clause "... and neither is the duplicate of an earlier entry". Earlier, we are told to "ignore later duplicates". If we are ignoring duplicates, how can we consider whether or not either "is the duplicate of an earlier entry"?

In other words, what would you expect the following to do?

```
<select id="selly">
  ...
  <option name="name3">name</option>
  <option name="mixed3">mixed name</option>
  <option id="mixed3" name="name3">mixed ID</option>
</select>

test(function () {
    assert_equals(selly.namedItem('mixed3')["value"], "mixed ID");
}, "if multiple items have a *name* or *id* value matching the parameter, return <mumble>");
```

In the above, the third element would be prefered according to Ryosuke's third paragraph. But since "name3" is a "duplicate of an earlier entry", the "id of an element" should not "[precede] its name". That is, the third element should not be considered, and the second should be returned.

When I ran the above test against WebKit, the third element was selected by ```namedItem```, but it's not clear to me if that's correct. Firefox returned the second element, but I suspect it's not for the reason I gave in the previous paragraph.